### PR TITLE
Initialize implicit-sdf without optimization

### PR DESCRIPTION
**Pull request by hbb1**
Hi, Thank you for the amazing repo!  I would like to improve the repo. 
1. SDF-based representation requires an initial shape. It is implemented by doing a few optimization steps now. 
Here I present an alternative implementation, which gives smooth initialization and  produces  comparable results. 

image

2. The customized operator `SpecifyGradient` in the SDS loss appears hack.  This reparameterization gives the equivalent gradient, and allows visualizing the loss scale during training. 
```python
# loss = SpecifyGradient.apply(latents, grad)
# not straghtforward, use a reparameterization trick instead 
target = (latents-grad).detach() # the target is the denoised latents
loss = 0.5 * F.mse_loss(latents, target, reduction='sum')  / batch_size
```
the gradient of the `latent` is `0.5*2(latents - (latents-grad)) = grad`

### CODE REVIEW

**Comment by bennyguo** [comment]
Hi @hbb1 ! Very nice work!

- For the SDF initialization, I think we still need the optimization-based method, because calculating analytic SDF is impossible if we want to initialize the shape according to a given mesh. So I think it's better that we keep the old `initialize_shape` function and add an `sdf_bias` (Union[float,str], if not float, in ['sphere', 'ellipsoid']) and an `sdf_bias_params` (dict) configuration. The sdf bias value is computed and added to the network-output SDF in `forward_X`. In this way we can choose which method to use for the initialization.
- The parametrization seems very reasonable. Please give me some time to test.

**Comment by thuliu-yt16** [comment]
Appreciate the contribution very much! 

For the parametrization, it seems to be able to merge without change. For the sdf initialization, I think keeping both initializations makes sense since we plan to have a mesh initialization later.  Please also consider using `threestudio.warn` in `threestudio/__init__.py` for logging, e.g. use `threestudio.warn("initialize_shape is deprecated, use get_init_sdf instead")`  if you need.

**Comment by bennyguo** [comment]
@hbb1 See https://github.com/threestudio-project/threestudio/pull/57 for the new PR, where I adapt the code based on my comments above. Thank you for the contribution!
